### PR TITLE
Update set_thetalim documentation

### LINKED ISSUE #16462: Allow wedges of polar plots to include theta = 0.
### Feature request

**Summary**

When creating wedges fro polar plots, it is not possible to create a wedge which include theta = 0.

**Code for reproduction**

```python
from matplotlib import pyplot as plt
from math import pi

fig = plt.figure()
ax = plt.axes(polar=True)

ax.scatter([i*pi/6 for i in range 12], [1]*12)
ax.set_thetalim(300, 60)

plt.show()
```

```python
from matplotlib import pyplot as plt
from math import pi

fig = plt.figure()
ax = plt.axes(polar=True)

ax.scatter([i*pi/6 for i in range 12], [1]*12)
ax.set_thetalim(60, 300)

plt.show()
```

**Actual outcome**

Both code snippets produce the same outcome, a plot ranging from 60 degrees to 300 degrees.

**Expected outcome**

When `theta_min < 360 and theta_max > 0` the produced wedge should range from `theta_min` to `theta_max`

### PR DESCRIPTION
**Pull request by dstansby**
Fixes #16462 - I think this is the best way to go, but suggestions welcome.

### CODE REVIEW

File: lib/matplotlib/projections/polar.py [hunk 1]
```diff
@@ -1010,12 +1010,13 @@ def set_thetalim(self, *args, **kwargs):
...
-            Minimum value in degrees.
-        thetamax : float
-            Maximum value in degrees.
+        Can take the following signatures:
+
+        - ``set_thetalim(minval, maxval)``: Set the limits in radians.
+        - ``set_thetalim(thetamin=minval, thetamax=maxval)``: Set the limits
+          in degrees.
+
+        where a and b are the minimum and maximum limits.
```
Comment (ImportanceOfBeingErnest) [inline lib/matplotlib/projections/polar.py]: where *minval* and *maxval* are ...

File: lib/matplotlib/projections/polar.py [hunk 2]
```diff
@@ -1010,12 +1010,13 @@ def set_thetalim(self, *args, **kwargs):
...
-        ----------
-        thetamin : float
-            Minimum value in degrees.
-        thetamax : float
-            Maximum value in degrees.
+        Can take the following signatures:
+
+        - ``set_thetalim(minval, maxval)``: Set the limits in radians.
+        - ``set_thetalim(thetamin=minval, thetamax=maxval)``: Set the limits
+          in degrees.
```
Comment (jklymak) [inline lib/matplotlib/projections/polar.py]: I think it would be nice to specify the wrapping around zero behaviour here?

Comment (dstansby) [reply lib/matplotlib/projections/polar.py]: Just checked, and wrapping behaviour is pretty wild! Should probably fix it elsewhere before documenting it...
```python
import numpy as np
import matplotlib.pyplot as plt


r = np.arange(0, 2, 0.01)
theta = 2 * np.pi * r

ax = plt.subplot(111, projection='polar')
ax.plot(theta, r)
ax.set_thetalim(0, 3 * np.pi)

plt.show()
```

Figure_1

Comment (jklymak) [reply lib/matplotlib/projections/polar.py]: ha ha, well even just that you can't specify around-zero behaviour with (320, 40)...

Comment (dstansby) [reply lib/matplotlib/projections/polar.py]: Ah cool, didn't know you could do that!

**Pull request merged**